### ISSUE TITLE
double free on _purple_connection_destroy ()

### ISSUE THREAD

**Issue by kradan** [majn/telegram-purple#283]
- pidgin version: Pidgin 2.10.12 (libpurple 2.10.12)
- telegram-purple version: HEAD (commit ee2a6fb740fe9580336e4af9a153b845bc715927)

Pidgin crashes on connection loss because of telegram purple.

```
Reading symbols from pidgin...(no debugging symbols found)...done.
(gdb) run
Starting program: /usr/bin/pidgin 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/i386-linux-gnu/libthread_db.so.1".
[New Thread 0xae6beb40 (LWP 31490)]
[New Thread 0xad851b40 (LWP 31547)]
[New Thread 0xad050b40 (LWP 31548)]
*** Error in `/usr/bin/pidgin': double free or corruption (!prev): 0x80c204c0 ***
======= Backtrace: =========
/lib/i386-linux-gnu/libc.so.6(+0x6929b)[0xb705029b]
/lib/i386-linux-gnu/libc.so.6(+0x6f527)[0xb7056527]
/lib/i386-linux-gnu/libc.so.6(+0x6fcd1)[0xb7056cd1]
/usr/lib/purple-2/telegram-purple.so(tgl_free_release+0x2a)[0xaf0907ca]
/usr/lib/purple-2/telegram-purple.so(tgls_clear_message+0x4b)[0xaf082f3b]
/usr/lib/purple-2/telegram-purple.so(tgls_free_message+0x21)[0xaf083131]
/usr/lib/purple-2/telegram-purple.so(tgls_free_message_gw+0x1d)[0xaf08318d]
/usr/lib/purple-2/telegram-purple.so(+0x6346c)[0xaf07f46c]
/usr/lib/purple-2/telegram-purple.so(tgl_free_all+0x81)[0xaf086ea1]
/usr/lib/purple-2/telegram-purple.so(connection_data_free+0xc1)[0xaf061581]
/usr/lib/purple-2/telegram-purple.so(+0x429ad)[0xaf05e9ad]
/usr/lib/libpurple.so.0(_purple_connection_destroy+0xdf)[0xb724da6f]
/usr/lib/libpurple.so.0(purple_account_disconnect+0x6c)[0xb7237bbc]
/usr/lib/libpurple.so.0(+0x419e4)[0xb724c9e4]
/lib/i386-linux-gnu/libglib-2.0.so.0(+0x47981)[0xb73ad981]
/lib/i386-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x14b)[0xb73acd8b]
/lib/i386-linux-gnu/libglib-2.0.so.0(+0x47179)[0xb73ad179]
/lib/i386-linux-gnu/libglib-2.0.so.0(g_main_loop_run+0xf9)[0xb73ad529]
/usr/lib/i386-linux-gnu/libgtk-x11-2.0.so.0(gtk_main+0xb5)[0xb78b3315]
/usr/bin/pidgin(main+0xd62)[0x800256e2]
/lib/i386-linux-gnu/libc.so.6(__libc_start_main+0xf7)[0xb6fff517]
/usr/bin/pidgin(+0x25a5a)[0x80025a5a]
======= Memory map: ========
80000000-80120000 r-xp 00000000 fe:00 132751     /usr/bin/pidgin
80120000-80125000 r--p 0011f000 fe:00 132751     /usr/bin/pidgin
80125000-80126000 rw-p 00124000 fe:00 132751     /usr/bin/pidgin
80126000-80e94000 rw-p 00000000 00:00 0          [heap]
aad00000-aad21000 rw-p 00000000 00:00 0 
aad21000-aae00000 ---p 00000000 00:00 0 
aaeab000-aaedc000 rw-p 00000000 00:00 0 
aaedc000-abedc000 rw-s 00000000 00:05 45023256   /SYSV00000000 (deleted)
abeff000-ac12d000 rw-p 00000000 00:00 0 
ac12d000-ac18b000 r-xp 00000000 fe:00 167288     /usr/lib/i386-linux-gnu/libjpeg.so.62.1.0
ac18b000-ac18c000 r--p 0005d000 fe:00 167288     /usr/lib/i386-linux-gnu/libjpeg.so.62.1.0
ac18c000-ac18d000 rw-p 0005e000 fe:00 167288     /usr/lib/i386-linux-gnu/libjpeg.so.62.1.0
ac230000-ac290000 rw-s 00000000 00:05 44957716   /SYSV00000000 (deleted)
ac290000-ac2d3000 r-xp 00000000 fe:00 3570       /lib/i386-linux-gnu/libnss_mymachines.so.2
ac2d3000-ac2d5000 r--p 00042000 fe:00 3570       /lib/i386-linux-gnu/libnss_mymachines.so.2
ac2d5000-ac2d6000 rw-p 00044000 fe:00 3570       /lib/i386-linux-gnu/libnss_mymachines.so.2
ac2d6000-ac2e6000 r-xp 00000000 fe:00 1409       /lib/i386-linux-gnu/libnss_myhostname.so.2
ac2e6000-ac2e7000 ---p 00010000 fe:00 1409       /lib/i386-linux-gnu/libnss_myhostname.so.2
ac2e7000-ac2e8000 r--p 00010000 fe:00 1409       /lib/i386-linux-gnu/libnss_myhostname.so.2
ac2e8000-ac2e9000 rw-p 00011000 fe:00 1409       /lib/i386-linux-gnu/libnss_myhostname.so.2
ac2e9000-ac349000 rw-s 00000000 00:05 44859396   /SYSV00000000 (deleted)
ac349000-ac3f5000 r--p 00000000 fe:00 274069     /usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf
ac3f5000-ac449000 rw-p 00000000 00:00 0 
ac449000-ac4fc000 r-xp 00000000 fe:00 152798     /usr/lib/i386-linux-gnu/libaspell.so.15.2.0
ac4fc000-ac4ff000 r--p 000b2000 fe:00 152798     /usr/lib/i386-linux-gnu/libaspell.so.15.2.0
ac4ff000-ac500000 rw-p 000b5000 fe:00 152798     /usr/lib/i386-linux-gnu/libaspell.so.15.2.0
ac500000-ac521000 rw-p 00000000 00:00 0 
ac521000-ac600000 ---p 00000000 00:00 0 
ac61e000-ac621000 rw-p 00000000 00:00 0 
ac621000-ac626000 r-xp 00000000 fe:00 130865     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jpeg.so
ac626000-ac627000 r--p 00004000 fe:00 130865     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jpeg.so
ac627000-ac628000 rw-p 00005000 fe:00 130865     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jpeg.so
ac628000-ac63a000 r--p 00000000 fe:00 265414     /usr/share/fonts/opentype/cantarell/Cantarell-Bold.otf
ac63a000-ac69f000 r-xp 00000000 fe:00 143896     /usr/lib/i386-linux-gnu/libhunspell-1.4.so.0.0.0
ac69f000-ac6a0000 r--p 00064000 fe:00 143896     /usr/lib/i386-linux-gnu/libhunspell-1.4.so.0.0.0
ac6a0000-ac6a4000 rw-p 00065000 fe:00 143896     /usr/lib/i386-linux-gnu/libhunspell-1.4.so.0.0.0
ac6a4000-ac6a6000 rw-p 00000000 00:00 0 
ac6a6000-ac6ab000 r-xp 00000000 fe:00 1038       /lib/i386-linux-gnu/libnss_dns-2.22.so
ac6ab000-ac6ac000 r--p 00004000 fe:00 1038       /lib/i386-linux-gnu/libnss_dns-2.22.so
ac6ac000-ac6ad000 rw-p 00005000 fe:00 1038       /lib/i386-linux-gnu/libnss_dns-2.22.so
ac6ad000-ac6af000 r-xp 00000000 fe:00 2530       /lib/i386-linux-gnu/libnss_mdns4_minimal.so.2
ac6af000-ac6b0000 r--p 00001000 fe:00 2530       /lib/i386-linux-gnu/libnss_mdns4_minimal.so.2
ac6b0000-ac6b1000 rw-p 00002000 fe:00 2530       /lib/i386-linux-gnu/libnss_mdns4_minimal.so.2
ac6b1000-ac6b3000 r-xp 00000000 fe:00 179224     /usr/lib/i386-linux-gnu/gconv/ISO8859-1.so
ac6b3000-ac6b4000 r--p 00001000 fe:00 179224     /usr/lib/i386-linux-gnu/gconv/ISO8859-1.so
ac6b4000-ac6b5000 rw-p 00002000 fe:00 179224     /usr/lib/i386-linux-gnu/gconv/ISO8859-1.so
ac6b5000-ac6b7000 r-xp 00000000 fe:00 143911     /usr/lib/i386-linux-gnu/enchant/libenchant_aspell.so
ac6b7000-ac6b8000 r--p 00001000 fe:00 143911     /usr/lib/i386-linux-gnu/enchant/libenchant_aspell.so
ac6b8000-ac6b9000 rw-p 00002000 fe:00 143911     /usr/lib/i386-linux-gnu/enchant/libenchant_aspell.so
ac6b9000-ac6c5000 r-xp 00000000 fe:00 152955     /usr/lib/i386-linux-gnu/enchant/libenchant_ispell.so
ac6c5000-ac6c6000 r--p 0000b000 fe:00 152955     /usr/lib/i386-linux-gnu/enchant/libenchant_ispell.so
ac6c6000-ac6c7000 rw-p 0000c000 fe:00 152955     /usr/lib/i386-linux-gnu/enchant/libenchant_ispell.so
ac6c7000-ac6cb000 r-xp 00000000 fe:00 152957     /usr/lib/i386-linux-gnu/enchant/libenchant_myspell.so
ac6cb000-ac6cc000 r--p 00003000 fe:00 152957     /usr/lib/i386-linux-gnu/enchant/libenchant_myspell.so
ac6cc000-ac6cd000 rw-p 00004000 fe:00 152957     /usr/lib/i386-linux-gnu/enchant/libenchant_myspell.so
ac6cd000-ac6d6000 r-xp 00000000 fe:00 143912     /usr/lib/i386-linux-gnu/enchant/libenchant_hspell.so
ac6d6000-ac6d7000 r--p 00008000 fe:00 143912     /usr/lib/i386-linux-gnu/enchant/libenchant_hspell.so
ac6d7000-ac6d9000 rw-p 00009000 fe:00 143912     /usr/lib/i386-linux-gnu/enchant/libenchant_hspell.so
ac6d9000-ac6ea000 r--p 00000000 fe:00 265417     /usr/share/fonts/opentype/cantarell/Cantarell-Regular.otf
ac6ea000-ac6ed000 r--s 00000000 fe:00 410791     /var/cache/fontconfig/6aa41aa22e18b8fa06a12da28ea9c28b-le32d4.cache-4
ac6ed000-ac6f5000 r--s 00000000 fe:00 396798     /var/cache/fontconfig/945677eb7aeaf62f1d50efc3fb3ec7d8-le32d4.cache-4
ac6f5000-ac6fa000 r--s 00000000 fe:00 397397     /var/cache/fontconfig/3047814df9a2f067bd2d96a2b9c36e5a-le32d4.cache-4
ac6fa000-ac700000 r--s 00000000 fe:00 401542     /var/cache/fontconfig/bf3b770c553c462765856025a94f1ce6-le32d4.cache-4
ac700000-ac721000 rw-p 00000000 00:00 0 
ac721000-ac800000 ---p 00000000 00:00 0 
ac800000-ac803000 r--s 00000000 fe:00 398683     /var/cache/fontconfig/de156ccd2eddbdc19d37a45b8b2aac9c-le32d4.cache-4
ac803000-ac806000 r--s 00000000 fe:00 397263     /var/cache/fontconfig/e49e89034d371f0f9de17aab02136486-le32d4.cache-4
ac806000-ac816000 r--s 00000000 fe:00 396963     /var/cache/fontconfig/d52a8644073d54c13679302ca1180695-le32d4.cache-4
ac816000-ac81f000 r--s 00000000 fe:00 393876     /var/cache/fontconfig/d589a48862398ed80a3d6066f4f56f4c-le32d4.cache-4
ac81f000-ac830000 r--s 00000000 fe:00 398684     /var/cache/fontconfig/e13b20fdb08344e0e664864cc2ede53d-le32d4.cache-4
ac830000-ac83a000 r--s 00000000 fe:00 396828     /var/cache/fontconfig/8f02d4cb045bd6ce15663e43f347c9f8-le32d4.cache-4
ac83a000-ac850000 r--s 00000000 fe:00 394186     /var/cache/fontconfig/467c019e582ee353435ea5c21d137ef6-le32d4.cache-4
ac850000-ac851000 ---p 00000000 00:00 0 
ac851000-ad051000 rw-p 00000000 00:00 0 
ad051000-ad052000 ---p 00000000 00:00 0 
ad052000-ad852000 rw-p 00000000 00:00 0 
ad852000-ad889000 r-xp 00000000 fe:00 155861     /usr/lib/i386-linux-gnu/gvfs/libgvfscommon.so
ad889000-ad88a000 ---p 00037000 fe:00 155861     /usr/lib/i386-linux-gnu/gvfs/libgvfscommon.so
ad88a000-ad88c000 r--p 00037000 fe:00 155861     /usr/lib/i386-linux-gnu/gvfs/libgvfscommon.so
ad88c000-ad88d000 rw-p 00039000 fe:00 155861     /usr/lib/i386-linux-gnu/gvfs/libgvfscommon.so
ad88d000-ad8c3000 r-xp 00000000 fe:00 162375     /usr/lib/i386-linux-gnu/gio/modules/libgvfsdbus.so
ad8c3000-ad8c4000 ---p 00036000 fe:00 162375     /usr/lib/i386-linux-gnu/gio/modules/libgvfsdbus.so
ad8c4000-ad8c5000 r--p 00036000 fe:00 162375     /usr/lib/i386-linux-gnu/gio/modules/libgvfsdbus.so
ad8c5000-ad8c6000 rw-p 00037000 fe:00 162375     /usr/lib/i386-linux-gnu/gio/modules/libgvfsdbus.so
ad8c6000-ad8e2000 r--p 00000000 fe:00 260720     /usr/share/icons/gnome/icon-theme.cache
ad8e2000-ad922000 r-xp 00000000 fe:00 142204     /usr/lib/i386-linux-gnu/libcroco-0.6.so.3.0.1
ad922000-ad923000 ---p 00040000 fe:00 142204     /usr/lib/i386-linux-gnu/libcroco-0.6.so.3.0.1
ad923000-ad925000 r--p 00040000 fe:00 142204     /usr/lib/i386-linux-gnu/libcroco-0.6.so.3.0.1
ad925000-ad926000 rw-p 00042000 fe:00 142204     /usr/lib/i386-linux-gnu/libcroco-0.6.so.3.0.1
ad926000-ad966000 r-xp 00000000 fe:00 141495     /usr/lib/i386-linux-gnu/librsvg-2.so.2.40.15
ad966000-ad967000 r--p 0003f000 fe:00 141495     /usr/lib/i386-linux-gnu/librsvg-2.so.2.40.15
ad967000-ad968000 rw-p 00040000 fe:00 141495     /usr/lib/i386-linux-gnu/librsvg-2.so.2.40.15
ad968000-ad969000 r--s 00000000 fe:00 395513     /var/cache/fontconfig/4794a0821666d79190d59a36cb4f44b5-le32d4.cache-4
ad969000-ad96b000 r--s 00000000 fe:00 397105     /var/cache/fontconfig/4b14b093aebc79c320de5e86ae1d3314-le32d4.cache-4
ad96b000-ad971000 r--p 00000000 fe:00 260665     /usr/share/icons/hicolor/icon-theme.cache
ad971000-ad98b000 r--p 00000000 fe:00 262257     /usr/share/icons/Adwaita/icon-theme.cache
ad98b000-ad9a5000 r--p 00000000 fe:00 159815     /usr/share/locale/de/LC_MESSAGES/glib20.mo
ad9a5000-ad9c5000 r--s 00000000 fe:00 264281     /usr/share/mime/mime.cache
ad9c5000-ada35000 r-xp 00000000 fe:00 148105     /usr/lib/i386-linux-gnu/nss/libfreebl3.so
ada35000-ada36000 ---p 00070000 fe:00 148105     /usr/lib/i386-linux-gnu/nss/libfreebl3.so
ada36000-ada37000 r--p 00070000 fe:00 148105     /usr/lib/i386-linux-gnu/nss/libfreebl3.so
ada37000-ada38000 rw-p 00071000 fe:00 148105     /usr/lib/i386-linux-gnu/nss/libfreebl3.so
ada38000-ada3c000 rw-p 00000000 00:00 0 
ada3c000-ada86000 r-xp 00000000 fe:00 148106     /usr/lib/i386-linux-gnu/nss/libsoftokn3.so
ada86000-ada87000 ---p 0004a000 fe:00 148106     /usr/lib/i386-linux-gnu/nss/libsoftokn3.so
ada87000-ada88000 r--p 0004a000 fe:00 148106     /usr/lib/i386-linux-gnu/nss/libsoftokn3.so
ada88000-ada89000 rw-p 0004b000 fe:00 148106     /usr/lib/i386-linux-gnu/nss/libsoftokn3.so
ada89000-adc73000 r-xp 00000000 fe:00 131170     /usr/lib/i386-linux-gnu/libdb-5.3.so
adc73000-adc74000 ---p 001ea000 fe:00 131170     /usr/lib/i386-linux-gnu/libdb-5.3.so
adc74000-adc79000 r--p 001ea000 fe:00 131170     /usr/lib/i386-linux-gnu/libdb-5.3.so
adc79000-adc7b000 rw-p 001ef000 fe:00 131170     /usr/lib/i386-linux-gnu/libdb-5.3.so
adc7b000-adc7c000 r--s 00000000 fe:00 398585     /var/cache/fontconfig/0bd3dc0958fa2205aaaa8ebb13e2872b-le32d4.cache-4
adc7c000-adc7e000 r-xp 00000000 fe:00 1974       /lib/i386-linux-gnu/libutil-2.22.so
adc7e000-adc7f000 r--p 00001000 fe:00 1974       /lib/i386-linux-gnu/libutil-2.22.so
adc7f000-adc80000 rw-p 00002000 fe:00 1974       /lib/i386-linux-gnu/libutil-2.22.so
adc80000-adc81000 r-xp 00000000 fe:00 134037     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
adc81000-adc82000 ---p 00001000 fe:00 134037     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
adc82000-adc83000 r--p 00001000 fe:00 134037     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
adc83000-adc84000 rw-p 00002000 fe:00 134037     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
adc84000-adc8e000 r--p 00000000 fe:00 170250     /usr/share/locale/de/LC_MESSAGES/gstreamer-1.0.mo
adc8e000-adc9c000 r-xp 00000000 fe:00 175641     /usr/lib/i386-linux-gnu/sasl2/libdigestmd5.so.2.0.25
adc9c000-adc9d000 r--p 0000d000 fe:00 175641     /usr/lib/i386-linux-gnu/sasl2/libdigestmd5.so.2.0.25
adc9d000-adc9e000 rw-p 0000e000 fe:00 175641     /usr/lib/i386-linux-gnu/sasl2/libdigestmd5.so.2.0.25
adc9e000-ade6d000 r-xp 00000000 fe:00 261046     /usr/lib/i386-linux-gnu/i686/cmov/libcrypto.so.1.0.2
ade6d000-ade6e000 ---p 001cf000 fe:00 261046     /usr/lib/i386-linux-gnu/i686/cmov/libcrypto.so.1.0.2
ade6e000-ade7e000 r--p 001cf000 fe:00 261046     /usr/lib/i386-linux-gnu/i686/cmov/libcrypto.so.1.0.2
ade7e000-ade87000 rw-p 001df000 fe:00 261046     /usr/lib/i386-linux-gnu/i686/cmov/libcrypto.so.1.0.2
ade87000-ade8a000 rw-p 00000000 00:00 0 
ade8a000-ade8c000 r--s 00000000 fe:00 394091     /var/cache/fontconfig/62f91419b9ebdb6975e7e41ab6412357-le32d4.cache-4
ade8c000-ade91000 r-xp 00000000 fe:00 130866     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
ade91000-ade92000 r--p 00004000 fe:00 130866     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
ade92000-ade93000 rw-p 00005000 fe:00 130866     /usr/lib/i386-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
ade93000-ade97000 r-xp 00000000 fe:00 175644     /usr/lib/i386-linux-gnu/sasl2/libplain.so.2.0.25
ade97000-ade98000 r--p 00003000 fe:00 175644     /usr/lib/i386-linux-gnu/sasl2/libplain.so.2.0.25
ade98000-ade99000 rw-p 00004000 fe:00 175644     /usr/lib/i386-linux-gnu/sasl2/libplain.so.2.0.25
ade99000-ade9f000 r-xp 00000000 fe:00 132526     /usr/lib/i386-linux-gnu/sasl2/libsasldb.so.2.0.25
ade9f000-adea0000 r--p 00005000 fe:00 132526     /usr/lib/i386-linux-gnu/sasl2/libsasldb.so.2.0.25
adea0000-adea1000 rw-p 00006000 fe:00 132526     /usr/lib/i386-linux-gnu/sasl2/libsasldb.so.2.0.25
adea1000-adea5000 r-xp 00000000 fe:00 175639     /usr/lib/i386-linux-gnu/sasl2/libanonymous.so.2.0.25
adea5000-adea6000 r--p 00003000 fe:00 175639     /usr/lib/i386-linux-gnu/sasl2/libanonymous.so.2.0.25
adea6000-adea7000 rw-p 00004000 fe:00 175639     /usr/lib/i386-linux-gnu/sasl2/libanonymous.so.2.0.25
adea7000-adeab000 r-xp 00000000 fe:00 175642     /usr/lib/i386-linux-gnu/sasl2/liblogin.so.2.0.25
adeab000-adeac000 r--p 00003000 fe:00 175642     /usr/lib/i386-linux-gnu/sasl2/liblogin.so.2.0.25
adeac000-adead000 rw-p 00004000 fe:00 175642     /usr/lib/i386-linux-gnu/sasl2/liblogin.so.2.0.25
adead000-adeb5000 r-xp 00000000 fe:00 175643     /usr/lib/i386-linux-gnu/sasl2/libntlm.so.2.0.25
adeb5000-adeb6000 r--p 00007000 fe:00 175643     /usr/lib/i386-linux-gnu/sasl2/libntlm.so.2.0.25
adeb6000-adeb7000 rw-p 00008000 fe:00 175643     /usr/lib/i386-linux-gnu/sasl2/libntlm.so.2.0.25
adeb7000-adebc000 r-xp 00000000 fe:00 175640     /usr/lib/i386-linux-gnu/sasl2/libcrammd5.so.2.0.25
adebc000-adebd000 r--p 00004000 fe:00 175640     /usr/lib/i386-linux-gnu/sasl2/libcrammd5.so.2.0.25
adebd000-adebe000 rw-p 00005000 fe:00 175640     /usr/lib/i386-linux-gnu/sasl2/libcrammd5.so.2.0.25
adebe000-adebf000 ---p 00000000 00:00 0 
adebf000-ae6bf000 rw-p 00000000 00:00 0 
ae6bf000-ae6c1000 r-xp 00000000 fe:00 139394     /usr/lib/purple-2/libyahoojp.so
ae6c1000-ae6c2000 r--p 00001000 fe:00 139394     /usr/lib/purple-2/libyahoojp.so
ae6c2000-ae6c3000 rw-p 00002000 fe:00 139394     /usr/lib/purple-2/libyahoojp.so
ae6c3000-ae6c6000 r-xp 00000000 fe:00 227832     /usr/lib/purple-2/libsnpp.so
ae6c6000-ae6c7000 r--p 00002000 fe:00 227832     /usr/lib/purple-2/libsnpp.so
ae6c7000-ae6c8000 rw-p 00003000 fe:00 227832     /usr/lib/purple-2/libsnpp.so
ae6c8000-ae6cb000 r-xp 00000000 fe:00 227824     /usr/lib/purple-2/slashexec.so
ae6cb000-ae6cc000 r--p 00002000 fe:00 227824     /usr/lib/purple-2/slashexec.so
ae6cc000-ae6cd000 rw-p 00003000 fe:00 227824     /usr/lib/purple-2/slashexec.so
ae6cd000-ae6ce000 r-xp 00000000 fe:00 227816     /usr/lib/purple-2/sslinfo.so
ae6ce000-ae6cf000 r--p 00001000 fe:00 227816     /usr/lib/purple-2/sslinfo.so
ae6cf000-ae6d0000 rw-p 00002000 fe:00 227816     /usr/lib/purple-2/sslinfo.so
ae6d0000-ae6d3000 r-xp 00000000 fe:00 227829     /usr/lib/purple-2/ignore.so
ae6d3000-ae6d4000 r--p 00002000 fe:00 227829     /usr/lib/purple-2/ignore.so
ae6d4000-ae6d5000 rw-p 00003000 fe:00 227829     /usr/lib/purple-2/ignore.so
ae6d5000-ae6d8000 r-xp 00000000 fe:00 227820     /usr/lib/purple-2/highlight.so
ae6d8000-ae6d9000 r--p 00002000 fe:00 227820     /usr/lib/purple-2/highlight.so
ae6d9000-ae6da000 rw-p 00003000 fe:00 227820     /usr/lib/purple-2/highlight.so
ae6da000-ae6e9000 r-xp 00000000 fe:00 139358     /usr/lib/purple-2/libgg.so
ae6e9000-ae6ea000 r--p 0000e000 fe:00 139358     /usr/lib/purple-2/libgg.so
ae6ea000-ae6eb000 rw-p 0000f000 fe:00 139358     /usr/lib/purple-2/libgg.so
ae6eb000-ae6ee000 r-xp 00000000 fe:00 1450       /lib/i386-linux-gnu/libcom_err.so.2.1
ae6ee000-ae6ef000 r--p 00002000 fe:00 1450       /lib/i386-linux-gnu/libcom_err.so.2.1
ae6ef000-ae6f0000 rw-p 00003000 fe:00 1450       /lib/i386-linux-gnu/libcom_err.so.2.1
ae6f0000-ae6ff000 r-xp 00000000 fe:00 145355     /usr/lib/i386-linux-gnu/libzephyr.so.4.0.0
ae6ff000-ae700000 rw-p 0000e000 fe:00 145355     /usr/lib/i386-linux-gnu/libzephyr.so.4.0.0
ae700000-ae707000 rw-p 00000000 00:00 0 
ae707000-ae709000 r--s 00000000 fe:00 397203     /var/cache/fontconfig/16326683038b281783a0ef8c680e3a10-le32d4.cache-4
ae709000-ae70a000 r-xp 00000000 fe:00 139338     /usr/lib/purple-2/buddynote.so
ae70a000-ae70b000 r--p 00000000 fe:00 139338     /usr/lib/purple-2/buddynote.so
ae70b000-ae70c000 rw-p 00001000 fe:00 139338     /usr/lib/purple-2/buddynote.so
ae70c000-ae70d000 r-xp 00000000 fe:00 227828     /usr/lib/purple-2/flip.so
ae70d000-ae70e000 r--p 00000000 fe:00 227828     /usr/lib/purple-2/flip.so
ae70e000-ae70f000 rw-p 00001000 fe:00 227828     /usr/lib/purple-2/flip.so
ae70f000-ae711000 r-xp 00000000 fe:00 227815     /usr/lib/purple-2/capsnot.so
ae711000-ae712000 r--p 00001000 fe:00 227815     /usr/lib/purple-2/capsnot.so
ae712000-ae713000 rw-p 00002000 fe:00 227815     /usr/lib/purple-2/capsnot.so
ae713000-ae716000 r-xp 00000000 fe:00 139336     /usr/lib/purple-2/autoaccept.so
ae716000-ae717000 r--p 00002000 fe:00 139336     /usr/lib/purple-2/autoaccept.so
ae717000-ae718000 rw-p 00003000 fe:00 139336     /usr/lib/purple-2/autoaccept.so
ae718000-ae723000 r-xp 00000000 fe:00 139344     /usr/lib/purple-2/log_reader.so
ae723000-ae724000 r--p 0000a000 fe:00 139344     /usr/lib/purple-2/log_reader.so
ae724000-ae725000 rw-p 0000b000 fe:00 139344     /usr/lib/purple-2/log_reader.so
ae725000-ae728000 r-xp 00000000 fe:00 227814     /usr/lib/purple-2/eight_ball.so
ae728000-ae729000 r--p 00002000 fe:00 227814     /usr/lib/purple-2/eight_ball.so
ae729000-ae72a000 rw-p 00003000 fe:00 227814     /usr/lib/purple-2/eight_ball.so
ae72a000-ae738000 r-xp 00000000 fe:00 139404     /usr/lib/purple-2/libzephyr.so
ae738000-ae739000 r--p 0000d000 fe:00 139404     /usr/lib/purple-2/libzephyr.so
ae739000-ae73a000 rw-p 0000e000 fe:00 139404     /usr/lib/purple-2/libzephyr.so
ae73a000-ae73e000 r-xp 00000000 fe:00 226654     /usr/lib/purple-2/autoreply.so
ae73e000-ae73f000 r--p 00003000 fe:00 226654     /usr/lib/purple-2/autoreply.so
Program received signal SIGABRT, Aborted.
0xb7fdadad in __kernel_vsyscall ()
(gdb) bt
#0  0xb7fdadad in __kernel_vsyscall ()
#1  0xb7012e26 in __GI_raise (sig=6) at ../sysdeps/unix/sysv/linux/raise.c:55
#2  0xb70143f7 in __GI_abort () at abort.c:89
#3  0xb70502a0 in __libc_message (do_abort=2, 
    fmt=0xb7147eac "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#4  0xb7056527 in malloc_printerr (action=<optimized out>, 
    str=0xb7147f1c "double free or corruption (!prev)", ptr=<optimized out>, 
    ar_ptr=0xb719b780 <main_arena>) at malloc.c:5007
#5  0xb7056cd1 in _int_free (av=0xb719b780 <main_arena>, p=<optimized out>, have_lock=0)
    at malloc.c:3868
#6  0xaf0907ca in tgl_free_release (ptr=0x80c204c0, size=78) at tools.c:162
#7  0xaf082f3b in tgls_clear_message (TLS=0x80aeca80, M=0x80d03400) at structures.c:2063
#8  0xaf083131 in tgls_free_message (TLS=0x80aeca80, M=0x80d03400) at structures.c:2090
#9  0xaf08318d in tgls_free_message_gw (M=0x80d03400, TLS=0x80aeca80) at structures.c:2462
#10 0xaf07f46c in tree_act_ex_message (T=<optimized out>, 
    act=act@entry=0xaf083170 <tgls_free_message_gw>, extra=extra@entry=0x80aeca80)
    at structures.c:84
#11 0xaf086ea1 in tgl_free_all (TLS=0x80aeca80) at structures.c:2471
#12 0xaf061581 in connection_data_free (conn=0x80aecef8) at tgp-structs.c:125
#13 0xaf05e9ad in tgprpl_close (gc=0x80666520) at telegram-purple.c:627
#14 0xb724da6f in _purple_connection_destroy () from /usr/lib/libpurple.so.0
#15 0xb7237bbc in purple_account_disconnect () from /usr/lib/libpurple.so.0
#16 0xb724c9e4 in ?? () from /usr/lib/libpurple.so.0
#17 0xb73ad981 in ?? () from /lib/i386-linux-gnu/libglib-2.0.so.0
#18 0xb73acd8b in g_main_context_dispatch () from /lib/i386-linux-gnu/libglib-2.0.so.0
#19 0xb73ad179 in ?? () from /lib/i386-linux-gnu/libglib-2.0.so.0
#20 0xb73ad529 in g_main_loop_run () from /lib/i386-linux-gnu/libglib-2.0.so.0
#21 0xb78b3315 in gtk_main () from /usr/lib/i386-linux-gnu/libgtk-x11-2.0.so.0
#22 0x800256e2 in main ()
```

**Comment by BenWiederhake** [comment]
Thanks for reminding me!
I already fixed that a while ago, and PR'ed it in the following places:
- https://github.com/vysheng/tgl/pull/94
- https://github.com/majn/tgl/pull/7

Please don't close this issue yet, as this isn't in mainstream telegram-purple; we first need to merge it into tgl.

**Comment by BenWiederhake** [comment]
Fixed in 1.3.0.  (I forgot to mention this issue in the commit message.)
